lib/utils/isNonNegativeInteger.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+// TODO: We need this polyfill because of the support of Node 10.
+//       When we will drop Node 10, please remove this polyfill.
+//       See <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger>
+const isInteger =
+	Number.isInteger ||
+	function (value) {
+		return typeof value === 'number' && isFinite(value) && Math.floor(value) === value;
+	};
+
+/**
+ * @param {unknown} value
+ */
+module.exports = function (value) {
+	return isInteger(value) && typeof value === 'number' && value >= 0;
```

Comment:
[note] `typeof value === 'number'` may be needless, but this is actually needed. Unless this `typeof`, a TypeScript error occurs:

```diff
-	return isInteger(value) && typeof value === 'number' && value >= 0;
+	return isInteger(value) && value >= 0;
```

```
lib/utils/isNonNegativeInteger.js:16:29 - error TS2571: Object is of type 'unknown'.

16  return isInteger(value) && value >= 0;
                               ~~~~~
```